programs/mango-v4/src/accounts_ix/openbook_v2_settle_funds.rs

```diff
@@ -0,0 +1,84 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token::{Token, TokenAccount};
+
+use crate::error::*;
+use crate::state::*;
+use openbook_v2::{
+    program::OpenbookV2,
+    state::{Market, OpenOrdersAccount},
+};
+
+#[derive(Accounts)]
+pub struct OpenbookV2SettleFunds<'info> {
+    #[account(
+        constraint = group.load()?.is_ix_enabled(IxGate::OpenbookV2SettleFunds) @ MangoError::IxIsDisabled,
+    )]
+    pub group: AccountLoader<'info, Group>,
+
+    #[account(
+        mut,
+        has_one = group,
+        constraint = account.load()?.is_operational() @ MangoError::AccountIsFrozen
+        // owner is checked at #1
+    )]
+    pub account: AccountLoader<'info, MangoAccountFixed>,
+
+    pub authority: Signer<'info>,
+
+    #[account(mut)]
+    pub open_orders: AccountLoader<'info, OpenOrdersAccount>,
+
+    #[account(
+        has_one = group,
+        has_one = openbook_v2_program,
+        has_one = openbook_v2_market_external,
+    )]
+    pub openbook_v2_market: AccountLoader<'info, OpenbookV2Market>,
+
+    pub openbook_v2_program: Program<'info, OpenbookV2>,
+
+    #[account(
+        mut,
+        constraint = openbook_v2_market_external.load()?.base_vault == market_base_vault.key(),
+        constraint = openbook_v2_market_external.load()?.quote_vault == market_quote_vault.key(),
+        constraint = openbook_v2_market_external.load()?.base_mint == base_bank.load()?.mint,
+        constraint = openbook_v2_market_external.load()?.quote_mint == quote_bank.load()?.mint,
```

Comment:
the last two checks are correct, but also unnecessary imo: if the mints don't match, the instruction would definitely fail.

Instead, we should check that the banks are for the expected token indexes stored in the openbook_v2_market (like for serum3, there it's done in the instruction code)